src/libraries/System.Threading.Overlapped/tests/ThreadPoolBoundHandle_AllocateNativeOverlappedTests.cs

```diff
@@ -124,24 +131,30 @@ public unsafe void AllocateNativeOverlapped_ObjectArrayAsPinData_DoesNotThrow()
     [PlatformSpecific(TestPlatforms.Windows)] // ThreadPoolBoundHandle.BindHandle is not supported on Unix
     public unsafe void AllocateNativeOverlapped_ObjectArrayWithNonBlittableTypeAsPinData_Throws()
     {
-        object[] array = new object[]
+        var array = new object[]
         {
             new NonBlittableType() { s = "foo" },
             new byte[5],
         };
-        using(ThreadPoolBoundHandle handle = CreateThreadPoolBoundHandle())
+
+        using (ThreadPoolBoundHandle handle = CreateThreadPoolBoundHandle())
         {
             AssertExtensions.Throws<ArgumentException>(null, () => handle.AllocateNativeOverlapped((_, __, ___) => { }, new object(), array));
+            AssertExtensions.Throws<ArgumentException>(null, () => handle.UnsafeAllocateNativeOverlapped((_, __, ___) => { }, new object(), array));
         }
     }
 
-    [Fact]
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
     [PlatformSpecific(TestPlatforms.Windows)] // ThreadPoolBoundHandle.BindHandle is not supported on Unix
-    public unsafe void AllocateNativeOverlapped_ReturnedNativeOverlapped_AllFieldsZero()
+    public unsafe void AllocateNativeOverlapped_ReturnedNativeOverlapped_AllFieldsZero(bool useUnsafe)
     {
-        using(ThreadPoolBoundHandle handle = CreateThreadPoolBoundHandle())
+        using (ThreadPoolBoundHandle handle = CreateThreadPoolBoundHandle())
         {
-            NativeOverlapped* overlapped = handle.AllocateNativeOverlapped((_, __, ___) => { }, new object(), new byte[256]);
+            NativeOverlapped* overlapped = useUnsafe ?
```

Comment:
One thing that we considered during the review was whether we should have a "flowExecutionContext" parameter instead of having an _Unsafe_ method vs. a normal capturing method. Using two methods seemed to better match what we do elsewhere in the SDK. But upon seeing this unit test code - do we think callers might want to toggle this based on a Boolean parameter? I'm tempted to say that unit tests are a unique scenario that don't match real-world app code, but your own thoughts would be valuable here as well.

Reply:
> Using two methods seemed to better match what we do elsewhere in the SDK

Yup